python/cudf/cudf/tests/test_json.py

```diff
@@ -327,3 +327,25 @@ def test_json_bad_protocol_string():
     got = cudf.read_json(test_string, lines=True)
 
     assert_eq(expect, got)
+
+
+def test_json_corner_case_with_escape_and_double_quote_char(tmpdir):
+    fname = tmpdir.mkdir("gdf_json").join("tmp_json_escape_double_quote")
+
+    pdf = pd.DataFrame(
+        {
+            "a": ['ab"cd', "?1sde", "rdsd"],
+            "b": ["234", "\\", "lstm"],
+            "c": ["aeiou", "sample", "json"],
+        }
+    )
+    pdf.to_json(fname, compression="infer", lines=True, orient="records")
+
+    df = cudf.read_json(
+        fname, compression="infer", lines=True, orient="records"
+    )
+    pdf = pd.read_json(
+        fname, compression="infer", lines=True, orient="records"
+    )
+
+    assert_eq(pdf, df)
```

Comment:
Please also include a test which verifies written file contents, as well as a test to verify specific json text can be parsed as an expected dataframe.

https://github.com/rapidsai/cudf/issues/6222

Reply:
I thought I understood #6222 but looks like I don't. The test here is using `pd.read_json` to verify the written file. What kind of verification is missing here?

Reply:
`x == g(f(x))` says nothing about `g(x)` or `f(x)`, aside stating that `f`'s domain maps uniquely to/from `g`'s domain.
- What does `f(x)` produce?
- Is `g(x)` able to produce expected results, given a specific `x` ?

As is, the code doesn't test `json`. It tests compatibility with Pandas's json writer (which could itself have bugs).

It's probably fine. but it's at least something to remember about the tests we're writing (most of our tests operate this way).

Reply:
The question is: Do we trust Pandas to be the source of truth for JSON writing?

Reply:
Unfortunately, our goal is very often to ensure compatibility with Pandas, bugs and all :)

Hm, in this case, we could use a string as input, instead of generating the JSON file using Pandas. Also, the output could be compared against a vector of strings, instead of Pandas output. @cwharris Do you think this is a better option here?

Reply:
>Do we trust Pandas to be the source of truth for JSON writing?
We do, but we don't have to, for this test.

Reply:
@vuule That would address my concerns, but I'd also want to keep the existing as an integration test. All are important, imo.

Reply:
> Unfortunately, our goal is very often to ensure compatibility with Pandas, bugs and all :)

Usually we don't want to replicate the bugs 😅

Reply:
> Usually we don't want to replicate the bugs 

Sorry for the bad joke. IIRC, we did consider replicating some of Pandas quirks at one point, but don't remember the details. There were open issues about Pandas and cuDF failing in a different way with invalid input :)